lightning/src/events/bump_transaction.rs

```diff
@@ -613,15 +620,28 @@ where
 		let total_input_amount = must_spend_amount +
 			coin_selection.confirmed_utxos.iter().map(|utxo| utxo.output.value).sum::<u64>();
 
-		self.process_coin_selection(&mut anchor_tx, coin_selection);
+		self.process_coin_selection(&mut anchor_tx, &coin_selection);
 		let anchor_txid = anchor_tx.txid();
 
+		// construct psbt
+		let mut anchor_psbt = PartiallySignedTransaction::from_unsigned_tx(anchor_tx.clone()).unwrap();
+		// add witness_utxo to anchor input
+		anchor_psbt.inputs[0].witness_utxo = Some(anchor_descriptor.previous_utxo());
+		// add witness_utxo to remaining inputs
+		for (idx, utxo) in coin_selection.confirmed_utxos.into_iter().enumerate() {
+			// add 1 to skip the anchor input
+			let index = idx + 1;
+			debug_assert_eq!(anchor_psbt.unsigned_tx.input[index].previous_output, utxo.outpoint);
+			anchor_psbt.inputs[index].witness_utxo = Some(utxo.output);
```

Comment:
> How do we know the given output was actually a segwit one (same goes for the HTLC transactions below)?

We don't but this is just providing the information we have, for non-segwit you're supposed to provide the full transaction to prevent attacks against signing for the wrong fee, however we don't have the information here and this should be sufficient since they aren't getting the PSBT from someone else.

> Further, if its taproot, don't we need more info than this?

This should be suffienct, they may need more info for their own inputs but they should have that in their wallet. Strictly speaking, this is more information than the previous api and provides the missing piece from before.

> And also redeem_script probably needs filling in.

We don't get that from the coin selection api and they user should have it anyways.

Reply:
> Right, but if we aren't including it in the PSBT, what's the point of the PSBT? I assume users can't just take the PSBT and hand it to some signer, it needs the redeem_script and other info.

Using bdk we can just call `sign_psbt` and it'll handle it for you. I believe bitcoin core's `walletprocesspsbt` will do the same.

Reply:
Okay changed for only segwit.